x-pack/test/api_integration/apis/management/index_management/indices.js

```diff
@@ -198,7 +198,10 @@ export default function({ getService }) {
           'ilm', // data enricher
           'isRollupIndex', // data enricher
         ];
-        expect(Object.keys(body[0])).to.eql(expectedKeys);
+        expectedKeys.map(key => {
+          expect(Object.keys(body[0])).to.contain(key);
+        });
+        expect(Object.keys(body[0]).length).to.equal(expectedKeys.length);
```

Comment:
I wonder how informative the output will be if these assertions fail. We'll know if the received keys _don't_ contain an _expected_ key, but we won't know if the received keys _do_ contain an _unexpected_ key (i.e. they have an extra key). I'm also not sure if I'd be able to act on a failure in the length assertion -- `expected 13 to equal `10` tells me there are unexpected keys in there, but not what they are.

The problem here seems to be the non-deterministic order of the keys. We can create determinism by consistently sorting the keys. What do you think of something like this?

```js
        const sortedExpectedKeys = [
          'health',
          'status',
          'name',
          'uuid',
          'primary',
          'replica',
          'documents',
          'size',
          'isFrozen',
          'aliases',
          // Cloud disables CCR, so wouldn't expect follower indices.
          'isFollowerIndex', // data enricher
          'ilm', // data enricher
          'isRollupIndex', // data enricher
        ].sort();

        const sortedReceivedKeys = Object.keys(body[0]).sort();

        // We need to sort the keys before comparing then, because race conditions
        // can cause enrichers to register in non-deterministic order.
        expect(sortedReceivedKeys).to.eql(sortedExpectedKeys);
```

Reply:
Hey, I like this solution. I like the reasoning as well. I will make the change.